src/web/UserPresence.ts

```diff
@@ -18,21 +18,24 @@ if (typeof(document) !== 'undefined') {
 
 export class UserPresence extends RX.UserPresence {
     private _isPresent: boolean;
+    private _isAppFocused: boolean;
```

Comment:
I'm concerned about adding another state variable here. The ifvisible module is already maintaining its notion of whether the app's main window is focused, and we're now maintaining a separate piece of state that tracks this. This is confusing and fragile.

Reply:
Removed using ifvisible module and implemented a Utils class AppVisibilityUtils that handles this now.